android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -537,23 +538,32 @@ private static ArrayList<HashMap<String, String>> getIMEList(Context context) {
     PackageManager packageManager = context.getPackageManager();
     for (InputMethodInfo imeInfo : imeList) {
       String id = imeInfo.getId();
+      String imeName = "";
       if (!id.startsWith(selfPackageName)) {
         // Attempt to get a readable name
         // Reference: https://stackoverflow.com/questions/62512501/get-human-readable-name-of-default-keyboard-not-package-name
         ComponentName componentName = ComponentName.unflattenFromString(id);
         if (componentName != null) {
           String packageName = componentName.getPackageName();
           try {
-            //PackageInfo info = packageManager.getPackageInfo(packageName, 0);
             ApplicationInfo info = packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA);
-            String imeName = (String)packageManager.getApplicationLabel(info);
+            imeName = (String)packageManager.getApplicationLabel(info);
+          } catch (PackageManager.NameNotFoundException e) {
+            // For Android 11+, this exception is thrown because we don't have QUERY_ALL_PACKAGES permission.
+            // We'll just display the package name instead.
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+              imeName = packageName;
+            } else {
+              KMLog.LogException(TAG, "Name not found", e);
+              continue;
+            }
```

Comment:
So, for earlier versions, you just won't list the input method? Why not use the same fallback?

Reply:
k, I can strike the `continue` and use the same fallback